net-gateway/src/endpoints/buffer/flush.rs

```diff
@@ -0,0 +1,83 @@
+use actix_multipart::Multipart;
+use actix_web::patch;
+use actix_web::web;
+use actix_web::HttpRequest;
+use futures::StreamExt;
+use net_core_api::api::envelope::envelope::Envelope;
+use net_core_api::api::result::result::ResultDTO;
+use net_core_api::core::decoder_api::Decoder;
+use net_token_verifier::fusion_auth::fusion_auth_verifier::FusionAuthVerifier;
+use net_updater_api::api::updaters::buffer_flush::buffer_flush_request::BufferFlushRequestDTO;
+use crate::core::quinn_client_endpoint_manager::QuinnClientEndpointManager;
+use crate::core::user_facing_error::UserFacingError;
+use crate::{authorization, config::Config};
+use net_core_api::core::typed_api::Typed;
+use net_core_api::core::encoder_api::Encoder;
+
+
+#[patch("/buffer")]
+async fn buffer(
+    config: web::Data<Config>,
+    req: HttpRequest,
+    mut payload: Multipart,
+) -> Result<&'static str, UserFacingError> {
+    //Auth stuff
+    let token_verifier = FusionAuthVerifier::new(
+        &config.fusion_auth_server_address.addr,
+        Some(config.fusion_auth_api_key.key.clone())
+    );
+
+    let authorization_result = authorization::authorize(
+        req,
+        Box::new(token_verifier)
+    ).await;
+
+    if authorization_result.is_err() { return Err(UserFacingError::Unauthorized); }
+    let token = authorization_result.unwrap();
+
+    let tenant_id = token.get_tenant_id();
+    if let Err(e) = tenant_id {
+        return Err(UserFacingError::InternalErrorWithDescription(e.to_string()));
+    }
+    let tenant_id = tenant_id.unwrap();
+    
+    let server_connection_result = QuinnClientEndpointManager::start_server_connection(
+        &config.quin_client_address.addr,
+        &config.quin_inserter.addr,
+        &config.quin_server_application.app,
+    ).await;
+    let mut server_connection = match server_connection_result {
+        Ok(server_connection) => server_connection,
+        Err(_) => return Err(UserFacingError::Timeout),
+    };
+
+    let mut delete = false;
+    if let Some(item) = payload.next().await {
+        if item.unwrap().name() == "delete" {
+            delete = true;
+        }
+    }
```

Comment:
How exactly should it work?

Reply:
If there is a `delete` item in the payload then we should just clear the buffer and not to transfer the data into the main table